pkgs/misc/emulators/dosbox-staging/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, lib, fetchFromGitHub, autoconf, autoconf-archive, automake, pkgconfig, alsaLib, SDL2, SDL2_net, libGL, libGLU, libogg, opusfile }:
+
+stdenv.mkDerivation rec {
+  pname = "dosbox-staging";
+  version = "v0.75.0";
+
+  src = fetchFromGitHub {
+    owner = "dreamer";
+    repo = pname;
+    rev = version;
+    sha256 = "00n5k2ya7ml55wvrk40p9il01f9k2q2b5g9b20n5zvbgv0d8c5ps";
+  };
+
+  hardeningDisable = [ "format" ];
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ autoconf autoconf-archive automake SDL2 libGL libGLU libogg opusfile alsaLib SDL2_net ];
+
+  preConfigure = "./autogen.sh";
+
+  preBuild = ''
+    buildFlagsArray=( "CXXFLAGS=-O3 -DNDEBUG" )
```

Comment:
Are these actually required? If so, wouldn't something like `buildFlags = [ "CXXFLAGS=-03" ];` be better?

Reply:
I believe `-DNDEBUG` is required, maybe @dreamer could elaborate why.

Reply:
I had trouble getting `buildFlags` to play nice with `-DNDEBUG` so I used `CXXFLAGS = "-O3 -DNDEBUG";` instead after I saw a few other packages using this.

Reply:
Both are required for release builds (in fact more compilation flags might be beneficial, with different sets per architecture, but simple `-O3 -DNDEBUG` is good enough across the board and good enough for the first release :)

Explanation:
- `NDEBUG` is standard POSIX-defined flag, that should be used for *all* C/C++ projects providing release binaries. [GNU doc](https://www.gnu.org/software/libc/manual/html_node/Consistency-Checking.html).
- `-O3` - the codebase is written in a way, that's very heavily affected by compiler optimizations; this level enables some crucial ones (like loop unrolling, switch/loop unrolling, vectorization, predictive commoning, etc)

One more note: dosbox-staging is mixed C/C++ codebase, so please use:
```
CFLAGS=-O3 -DNDEBUG
CXXFLAGS=-O3 -DNDEBUG
```

Reply:
@JoshuaFern I should have been more specific... I'm wondering if these values are needed **in the derivation** because maybe they are already implied by our build system. I don't know and this area is certainly a weak point in my knowledge. You could dig into our docs, builder code, or ping one of many people who know off the top of their head. Sorry I'm not being more help :disappointed:

Reply:
Running the build and searching through the log I'm not seeing -DNDEBUG and it's using -02 by default. I believe we need these manually defined. 

> Sorry I'm not being more help
You have good suggestions, thanks! :)